### ISSUE TITLE
Create a 1.23 branch?

### ISSUE THREAD

**Issue by atroy** [docker-php/docker-php#192]
Hi Joel,

I am currently fixing some api issues - regarding 1.22 (I create a PR later this week), but I was wondering if we should create a 1.23 branch according to [https://docs.docker.com/engine/reference/api/docker_remote_api_v1.23/](url)

Thanks in advance,
Alex

**Comment by joelwurtz** [comment]
Yes, i plan to make the update for 1.11 soon, in the meantime, master will become 1.23, so if you want to do some works on 1.23 it just base your PR on the master branch

**Comment by atroy** [comment]
Thanks for that - I will target master in this case.

**Comment by jmauerhan** [comment]
Any ETA on when this will be released? I'm blocked on a project because I can't use the 1.23 API.

**Comment by atroy** [comment]
@jmauerhan Hi, my team and I also working with Docker 1.23 API and it works perfektly fine (even with swarm). Have you already tried to use? Sure 1.22 is not 1.23 but there is not a big difference. Changes listed here https://docs.docker.com/engine/reference/api/docker_remote_api/#v1-23-api-changes

**Comment by jmauerhan** [comment]
I needed to specify ports when creating a container, and that was not working. I ended up just using the code to create the connection to docker from this library, and then doing the endpoints with my own code - which actually simplified my code, so I'm happy with it.

**Comment by atroy** [comment]
I fixed some issues regarding ports try use "docker-php/docker-php": "dev-master#4ea4afd03008f41894582e69fe470d060e616bdf" 

This code is currently running, with this version:
```php
$containerConfig = new ContainerConfig();
$containerConfig->setImage("nginx");

$hostConfig = new HostConfig();

$hostPort80Binding = new PortBinding();
$hostPort80Binding->setHostIp("0.0.0.0");
$hostPort80Binding->setHostPort("80");

$mapPorts = new ArrayObject();
$mapPorts["80/tcp"] = [$hostPort80Binding];
$hostConfig->setPortBindings($mapPorts);

$containerConfig->setHostConfig($hostConfig);
```

**Comment by joelwurtz** [comment]
1.23 branch created, i wll close this if anything don't work for 1.23 please make a new issue :)
